src/transaction_builder.js

```diff
@@ -676,6 +676,12 @@ TransactionBuilder.prototype.sign = function (vin, keyPair, redeemScript, hashTy
   var signed = input.pubKeys.some(function (pubKey, i) {
     if (!kpPubKey.equals(pubKey)) return false
     if (input.signatures[i]) throw new Error('Signature already exists')
+    if (!keyPair.compressed && (
+      input.prevOutType === scriptTypes.P2WPKH ||
+      input.prevOutType === scriptTypes.P2WSH ||
+      input.redeemScriptType === scriptTypes.P2WPKH ||
+      input.redeemScriptType === scriptTypes.P2WSH
+    )) throw new Error('BIP143 rejects uncompressed public keys in P2WPKH or P2WSH')
```

Comment:
@afk11 not an amazing place to put this,  but I intend to refactor this area hopefully soon